uqcsbot/starboard.py

```diff
@@ -85,11 +229,45 @@ def _create_sb_embed(self, recv: discord.Message) -> discord.Embed:
                 icon_url=recv.reference.resolved.author.display_avatar.url
             )
 
-            replied.set_footer(text=recv.reference.resolved.created_at.strftime('%b %d, %H:%M:%S'))
+            replied.set_footer(
+                text=recv.reference.resolved.created_at.astimezone(tz=self.BRISBANE_TZ).strftime('%b %d, %H:%M:%S')
+            )
 
             return [replied, embed]
-        
         return [embed]
+
+    async def _get_message_pair(self, channel, message_id: int) -> List[discord.Message]:
+        message = await channel.fetch_message(message_id)
+
+        if channel == self.starboard_channel:
+            sent = message
+            recv = await self._query_og_message(message_id)
+        else:
+            recv = message
+            sent = await self._query_sb_message(message_id)
+        
+        return [recv, sent]
+
+    @app_commands.command()
+    @app_commands.default_permissions(manage_messages=True)
+    async def cleanup_starboard(self, interaction: discord.Interaction):
+        """ Cleans up the last 100 messages from the starboard.
+        Removes any uqcsbot message that doesn't have a corresponding message id in the db, regardless of recv. """
+        sb_messages = await self.starboard_channel.history(limit=100)
```

Comment:
It appears that .history() returns an async generator, so await won't work here.

```
TypeError: object async_generator can't be used in 'await' expression
```

Reply:
"async iterators" are hard